src/compiler/core.ts

```diff
@@ -788,49 +897,71 @@ namespace ts {
     }
 
     /**
-     * Enumerates the properties of a Map<T>, invoking a callback and returning the first truthy result.
-     *
-     * @param map A map for which properties should be enumerated.
-     * @param callback A callback to invoke for each property.
+     * Array of every key in a map.
+     * May not actually return string[] if numbers were put into the map.
      */
-    export function forEachProperty<T, U>(map: Map<T>, callback: (value: T, key: string) => U): U {
-        let result: U;
-        for (const key in map) {
-            if (result = callback(map[key], key)) break;
-        }
-        return result;
+    export function keysOfMap<T>(map: Map<T>): string[] {
+        const keys: string[] = [];

```

Comment:
My [test](https://github.com/andy-ms/perf/blob/master/keysOfMap.js) shows it being slightly more efficient with a closure.